--- gajim-source/mac/gajim.spec	2024-05-22 21:33:20.307106129 +0200
+++ gajim-source/mac/gajim.spec	2025-12-20 00:00:06.273901355 +0100
@@ -11,9 +11,14 @@
 }
 
 import sys
+import glob
+
 sys.path.insert(0, os.path.join(cwd))
-from gajim.common.modules import MODULES
-hiddenimports = ['gajim.common.modules.' + m for m in MODULES]
+
+modules = glob.glob("gajim/common/modules/*.py")
+modules_list = [os.path.basename(f)[:-3] for f in modules if not f.endswith("__init__.py")]
+hiddenimports = ['gajim.common.modules.' + m for m in modules_list]
+
 sys.path.pop(0)
 
 a = Analysis(['launch.py'],
@@ -23,7 +28,7 @@
              hiddenimports=hiddenimports,
              hookspath=[],
              runtime_hooks=[],
-             excludes=[],
+             excludes=['docutils'],
              win_no_prefer_redirects=False,
              win_private_assemblies=False,
              cipher=block_cipher,
